python/cudf_polars/cudf_polars/experimental/benchmarks/tpch.py

```diff
@@ -6,24 +6,212 @@
 from __future__ import annotations
 
 import argparse
+import dataclasses
+import importlib
+import json
 import os
+import sys
 import time
-from datetime import date
-from typing import Any
+from collections import defaultdict
+from datetime import date, datetime, timezone
+from typing import TYPE_CHECKING, Any
 
 import numpy as np
+import pynvml
 
 import polars as pl
 
 from cudf_polars.dsl.translate import Translator
 from cudf_polars.experimental.parallel import evaluate_streaming
 
+if TYPE_CHECKING:
+    import pathlib
+
 # Without this setting, the first IO task to run
 # on each worker takes ~15 sec extra
 os.environ["KVIKIO_COMPAT_MODE"] = os.environ.get("KVIKIO_COMPAT_MODE", "on")
 os.environ["KVIKIO_NTHREADS"] = os.environ.get("KVIKIO_NTHREADS", "8")
 
 
+@dataclasses.dataclass
+class Record:
+    """Results for a single run of a single TPCH query."""
+
+    query: int
+    duration: float
+
+
+@dataclasses.dataclass
+class VersionInfo:
+    """Information about the version of the software used to run the query."""
+
+    version: str
+    commit: str | None
+
+
+@dataclasses.dataclass
+class PackageVersions:
+    """Information about the versions of the software used to run the query."""
+
+    cudf_polars: VersionInfo
+    polars: VersionInfo
+    python: VersionInfo
+    rapidsmpf: VersionInfo | None
+
+    @classmethod
+    def collect(cls) -> PackageVersions:
+        """Collect the versions of the software used to run the query."""
+        packages = [
+            "cudf_polars",
+            "polars",
+        ]
+        versions = {}
+        for name in packages:
+            package = importlib.import_module(name)
+            versions[name] = VersionInfo(
+                version=package.__version__,
+                commit=getattr(package, "__git_commit__", None),
```

Comment:
This always seems to be an empty string locally - Under what conditions does this return something useful?

Reply:
Looks like `cudf_polars` doesn't set it. It was for `cudf` and `rmm`, but we aren't capturing those.

I can just remove this.